app/src/main/java/net/squanchy/typeface/TypefaceController.kt

```diff
@@ -0,0 +1,26 @@
+package net.squanchy.typeface
+
+import android.graphics.Typeface
+import android.text.Spannable
+import android.text.SpannableString
+import android.text.Spanned
+
+class TypefaceController {
+
+    fun applyTypeface(text: CharSequence, typeface: Typeface) =
+            if (text !is Spannable) {
```

Comment:
Prefer positive checks: 

```kotlin
if (text is Spannable) {
  //...
} else {
  SpannableString(text).applyTypeface(typeface) // recurse here
}
```

At that point you can even inline `addSpanTo` into the positive branch since it's a one-liner and you're not adding any clarity by extracting it